source/material_model/rheology/visco_plastic.cc

```diff
@@ -0,0 +1,893 @@
+                                                                                       pressure_for_plasticity,
+                                                                                       drucker_prager_parameters.max_yield_stress);
+
+            // Step 4b: select if yield viscosity is based on Drucker Prager or stress limiter rheology
+            double viscosity_yield = viscosity_pre_yield;
+            switch (yield_mechanism)
+              {
+                case stress_limiter:
+                {
+                  //Step 4b-1: always rescale the viscosity back to the yield surface
+                  const double viscosity_limiter = yield_stress / (2.0 * ref_strain_rate)
+                                                   * std::pow((edot_ii/ref_strain_rate),
+                                                              1./exponents_stress_limiter[j] - 1.0);
+                  viscosity_yield = 1. / ( 1./viscosity_limiter + 1./viscosity_pre_yield);
+                  break;
+                }
+                case drucker_prager:
+                {
+                  // Step 4b-2: if the current stress is greater than the yield stress,
+                  // rescale the viscosity back to yield surface
+                  // If this is the fault material and rate-and-state friction is used,
+                  // assume that we are always yielding
+                  //std::cout << "j is: " << j << ", while fault_material_index +1 : " << friction_options.fault_composition_index +1 << std::endl;
+                  //std::cout << "current_stress - yield_stress is: " << current_stress - yield_stress  << " in composition " << j<< std::endl << std::endl;
+                  //                   AssertThrow(volume_fractions.size() == in.composition[0].size()+1, ExcMessage("there is a field too much...."));
+
+                  if ((current_stress >= yield_stress) |
+                      ((friction_options.get_use_theta()) && (j== friction_options.fault_composition_index + 1)))
+                    {
+                      // std::cout << "so we entered the yielding loop!" << std::endl << std::endl;
+                      viscosity_yield = drucker_prager_plasticity.compute_viscosity(current_cohesion,
+                                                                                    output_parameters.current_friction_angles[j],
+                                                                                    pressure_for_plasticity,
+                                                                                    current_edot_ii,
+                                                                                    drucker_prager_parameters.max_yield_stress,
+                                                                                    viscosity_pre_yield);
+                      output_parameters.composition_yielding[j] = true;
+                    }
+                  break;
+                }
+                case tresca:
```

Comment:
I don't think this tresca implementation is correct. The yield stress is calculated above with the drucker-prager criterion, but this should be the fault_strength used here. We switch out the DP criterion for the one from the benchmark, not use both of them somehow.